### PR TITLE
convert cgen.c to D

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/cgen.d [hunk 1]
```diff
@@ -131,88 +116,84 @@ code *cat(code *c1,code *c2)
 
 code *gen(code *c,code *cs)
 {
-#ifdef DEBUG                            /* this is a high usage routine */
```
Comment (thewilsonator) [inline src/dmd/backend/cgen.d]: Retain the comment.

Comment (WalterBright) [reply src/dmd/backend/cgen.d]: The comment is obsolete. The CodeBuilder.gen() retains the comment. The older gen() is rarely called.

**Comment by WalterBright** [comment]
The semaphoreci thing must have been a heisenbug. Sigh.

**Comment by WalterBright** [comment]
Thanks for your help with this. I appreciate it.

**Pull request merged**